source/xep/omemo/lurch/lurch.c

```diff
@@ -2674,3 +2700,68 @@ PURPLE_INIT_PLUGIN(lurch, lurch_plugin_init, info)
 
 #endif
 
+static void lurch_pep_muc_user_handler(JabberStream * js_p, const char * from, const char * real_jid)
+{
+  char ** split = (void *) 0;
+  const char * room_name = (void *) 0;
+  const char * buddy_nick = (void *) 0;
+  PurpleConversation * conv_p = (void *) 0;
+  JabberChat * muc_p = (void *) 0;
+  JabberChatMember * muc_member_p = (void *) 0;
+  int ret_val = 0;
+  char * err_msg_dbg = (void *) 0;
+  char * uname = (void *) 0;
+  char * db_fn_omemo = (void *) 0;
+
+  //TODO: Add status management
+
+  uname = lurch_util_uname_strip(purple_account_get_username(purple_connection_get_account(js_p->gc)));
+  db_fn_omemo = lurch_util_uname_get_db_fn(uname, LURCH_DB_NAME_OMEMO);
+
+  split = g_strsplit(from, "/", 2);
+  room_name = split[0];
+  buddy_nick = split[1];
+
+  ret_val = omemo_storage_chatlist_save(room_name, db_fn_omemo);
+  if (ret_val) {
+    err_msg_dbg = g_strdup_printf("Failed to look up %s in DB %s.", room_name, db_fn_omemo);
+    goto cleanup;
+  }
+
+  conv_p = purple_find_conversation_with_account(PURPLE_CONV_TYPE_CHAT, room_name, NULL);
+
+  if (!conv_p) {
+      conv_p = purple_conversation_new(PURPLE_CONV_TYPE_CHAT, purple_connection_get_account(js_p->gc), room_name);
+  }
+
+  muc_p = jabber_chat_find_by_conv(conv_p);
+  if (!muc_p) {
+    err_msg_dbg = g_strdup_printf("could not find muc struct for groupchat %s", room_name);
+    goto cleanup;
+  }
+
+  muc_member_p = g_hash_table_lookup(muc_p->members, buddy_nick);
+
+  if(!muc_member_p)
+  {
+    muc_member_p = g_malloc(sizeof(JabberChatMember));
+    muc_member_p->jid = g_strdup(real_jid);
+    muc_member_p->handle = "";
+    g_hash_table_insert(muc_p->members, g_strdup(buddy_nick), muc_member_p);
+  }
+  else
+  {
+    g_free(muc_member_p->jid);
+    muc_member_p->jid = g_strdup(real_jid);
+  }
+
+  cleanup:
+  if (err_msg_dbg) {
+    purple_conv_present_error(from, purple_connection_get_account(js_p->gc), LURCH_ERR_STRING_DECRYPT);
+    purple_debug_error("lurch", "%s: %s (%i)\n", __func__, err_msg_dbg, ret_val);
+    g_free(err_msg_dbg);
+  }
+  g_strfreev(split);
+  g_free(db_fn_omemo);
+  g_free(uname);
+}
```

Comment:
also this code. Is this from an updated lurch.c?

Reply:
No. I looked at what is needed and I added it

Reply:
Hm, there must be an other way to solve this. As stated, the goal is to not alter the lurch.c any more in the future and use the lurch.c source 'as is'. This will also allow quick updates of lurch with future releases.

Reply:
Initially I did differently adding list of couples (nickname, jid) to room entry in RosterItem but because it needs to be processed in lurch.c, it needs  a C/C++ adapter and also specific code in lurch. I found it cleaner to do it this way.